expression/builtin_miscellaneous_vec.go

```diff
@@ -334,20 +334,22 @@ func (b *builtinSleepSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) e
 }
 
 func doSleep(secs float64, sessVars *variable.SessionVars) (isKilled bool) {
+	if secs <= 0.0 {
+		return false
+	}
 	dur := time.Duration(secs * float64(time.Second.Nanoseconds()))
 	ticker := time.NewTicker(10 * time.Millisecond)
 	defer ticker.Stop()
-	start := time.Now()
+	timer := time.NewTimer(dur)
+	defer timer.Stop()
```

Comment:
If the `timer` is fired, it needn't to stop.
So we can only stop it when killed